app/services/search/user_agent_generator.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Search
+  module UserAgentGenerator
```

Comment:
This extraction is not idiomatic Ruby 😅 FFor concerns, the naming is not Ruby-ish enough. Such modules are usually named with the prefix `has` (e.g., `hasSettings`) or the suffix `able` (e.g., `include Disardable`). You can read mode about it in this article: https://www.akshaykhot.com/how-rails-concerns-work-and-how-to-use-them/.

We can also make the argument that this looks more like a case for a helper or PORO:

```ruby
class UserAgentHeaderGenerator
    USER_AGENT_HEADER = 'User-Agent'

    def initialize
			@header_name = USER_AGENT_HEADER
    end

		def call
				{ @header_name => Ronin::Web::UserAgents.chrome.random }
    end
end
```

Then in the service classes:

```ruby
headers = {}
user_agent_header = UserAgentHeaderGenerator.call()

response = Faraday.get(uri, nil, headers.merge(user_agent_header))
```

> **Note**
> When a class uses the method `call`, we can use this syntax `UserAgentHeaderGenerator.()`. This is any FYI, not recommended though 😅